Sources/Confidence/Confidence.swift

```diff
@@ -232,53 +199,117 @@ public class Confidence: ConfidenceEventSender {
         var reconciledCtx = parentContext.filter {
             !removedContextKeys.contains($0.key)
         }
-        self.contextSubject.value.forEach { entry in
+        self.context.forEach { entry in
             reconciledCtx.updateValue(entry.value, forKey: entry.key)
         }
         return reconciledCtx
     }
 
-    public func putContext(key: String, value: ConfidenceValue) {
-        withLock { confidence in
-            var map = confidence.contextSubject.value
+    public func putContext(key: String, value: ConfidenceValue) async {
+        await withSemaphoreAsync { [weak self] in
+            guard let self = self else {
+                return
+            }
+            var map = self.context
             map[key] = value
-            confidence.contextSubject.value = map
-            confidence.debugLogger?.logContext(action: "PutContext", context: confidence.contextSubject.value)
+            context = map
+            do {
+                try await self.fetchAndActivate()
+                debugLogger?.logContext(action: "PutContext", context: context)
+            } catch {
+                debugLogger?.logMessage(message: "Error when putting context: \(error)", isWarning: true)
+            }
         }
     }
 
-    public func putContext(context: ConfidenceStruct) {
-        withLock { confidence in
-            var map = confidence.contextSubject.value
+
+    public func putContextLocal(context: ConfidenceStruct, removeKeys removedKeys: [String] = []) {
+        withSemaphore { [weak self] in
```

Comment:
This semaphore might be slow to unlock due to network calls. This function, which looks immediate and sync, could actually be very slow